### ISSUE TITLE
[Accessibility] [Dropdown] [Expandable Menu with item auto expanded] > Items Dropdown] > Missing and incorrect ARIA roles and attributes for menu structure and nested items

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#2023]
[A11Y-615](https://jira.cengage.com/browse/A11Y-615)

**Environment:**
Windows 11 / Chrome 137 / JAWS 2025
Windows 11 / Edge 136 / NVDA 2025
MacOS 15 / Safari / VoiceOver

**Steps to Reproduce:**
1. Open the URL - https://react-magma.cengage.com/version/4.9.1/api/dropdown/
2. Turn on a screen reader (JAWS/NVDA/VoiceOver).
3. Navigate through the menu and nested items using keyboard or screen reader navigation.
4. Inspect the DOM structure and ARIA roles applied to the menu and its items.

**Actual Result:**

The parent container of the menuitems(paste,prosciutto and pizza) lacks role="menu".
Expand/collapse buttons (Paste, Prosciutto) do not have role="menuitem" or aria-haspopup="true".
Nested menu items ( Fresh, Processed, Domestic, Speck) are not wrapped in a container with role="menu".
role="group" is incorrectly applied to a non-grouping container.

Refer to the screencast:
https://app.screencast.com/DnMcCSQDUOYNi
https://app.screencast.com/7UOp2ljalQqXa

Instance: This issue is happening for 'Text only Expandable Menu Item,' sub component as well.

**Expected Result:**

The parent container should have role="menu" and be properly labeled using aria-labelledby.
Expand/collapse buttons should have role="menuitem", aria-haspopup="true", aria-expanded, and aria-controls.
Nested menu items should be grouped within a container with role="menu" and labeled appropriately.
role="group" should be removed unless grouping form controls.

**User Impact:**
Screen reader users may not understand the structure or behavior of the menu and its nested items. This can lead to confusion, incorrect navigation, and difficulty accessing submenu content.

WCAG Guidelines:
4.1.2 Name, Role, Value (Level A)

**Bug Fix Recommendation:**

Add role="menu" to the parent container of the menu (`<div aria-labelledby="c4081554-223d-4f80-9c8d-328ebf1e528b">`)
Add role="menuitem" and aria-haspopup="true" to expandable buttons, along with aria-expanded and aria-controls ( Paste, Prosciutto button)
Wrap nested menu items in a container with role="menu" and appropriate labeling.(parent container for Fresh, Processed menuitem and the parent container for Domestic and Speck menuitem)
Remove role="group" from non-semantic containers.

**Comment by olehnoskov-cengage** [comment]
This [[Accessibility] [Global] [Dropdown] [Basic Usage]> Dropdown button should include aria-controls to reference the associated menu](https://github.com/cengage/react-magma/issues/2024) ticket can be fixed in the scope of the current one.

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/dropdown--expandable-items
V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/dropdown--expandable-items

Everything works as expected. No bugs found.
